packages/powersync_core/lib/src/exceptions.dart

```diff
@@ -58,12 +58,7 @@ class SyncResponseException implements Exception {
       final details = _stringOrFirst(decoded['error']?['details']) ?? body;
       final message = '${response.reasonPhrase ?? "Request failed"}: $details';
       return SyncResponseException(response.statusCode, message);
-    } on FormatException catch (_) {
```

Comment:
This format exception looks correct to me, but I'm fine catching all `Exception`s as well.

I am wondering if we should refactor the common part of this into a say `static SyncResponseException fromResponseBody(String body)` though?